sphinx_gallery/docs_resolv.py

```diff
@@ -232,8 +233,15 @@ def _get_link(self, cobj):
                 fname_idx = value[cobj['name']][0]
 
         if fname_idx is not None:
-            fname = self._searchindex['filenames'][fname_idx] + '.html'
-
+            fname = self._searchindex['filenames'][fname_idx]
+            try:
+                import sphinx
+            except Exception:
+                fname = os.path.splitext(fname)[0]
+            else:
+                if LooseVersion(sphinx.__version__) >= LooseVersion('1.5'):
```

Comment:
Bonus points if you add a comment to clarify why the work-around for 1.5 is needed.